source/python/convert_intrinsics_opencv_to_halcon.py

```diff
@@ -0,0 +1,297 @@
+                f"",
+                f"ImageWidth:imgw:  {self.ImageSize[0]};",
+                f"  INT:1:32768;",
+                f'  "Width of the images";',
+                f"",
+                f"ImageHeight:imgh:  {self.ImageSize[1]};",
+                f"  INT:1:32768;",
+                f'  "Height of the images";',
+            ]
+        )
+
+    def _write_to_dat_file(self, filepath: Path) -> None:
+        if not exists(filepath):
+            with open(filepath, "w") as text_file:
+                text_file.write(self._get_dat_text())
+        else:
+            raise Exception(f"{filepath} already exists!")
+
+
+@dataclass
+class ZividCameraParameters:
+    pixel_size: NDArray[float]
+    image_size: NDArray[int]
+
+    @staticmethod
+    def zivid_one() -> ZividCameraParameters:
+        return ZividCameraParameters(pixel_size=5.86e-6, image_size=[1920, 1200])
+
+    @staticmethod
+    def zivid_one_plus() -> ZividCameraParameters:
+        return ZividCameraParameters.zivid_one()
+
+    @staticmethod
+    def zivid_two() -> ZividCameraParameters:
+        return ZividCameraParameters(pixel_size=4.50e-6, image_size=[1944, 1200])
+
+    @staticmethod
+    def from_name(name: str) -> ZividCameraParameters:
+        if name.lower() == "zivid one":
+            return ZividCameraParameters.zivid_one()
+        if name.lower() == "zivid one plus":
+            return ZividCameraParameters.zivid_one_plus()
+        if name.lower() == "zivid one plus s":
+            return ZividCameraParameters.zivid_one_plus()
+        if name.lower() == "zivid one plus m":
+            return ZividCameraParameters.zivid_one_plus()
+        if name.lower() == "zivid one plus l":
+            return ZividCameraParameters.zivid_one_plus()
+        if name.lower() == "zivid two":
+            return ZividCameraParameters.zivid_two()
+        raise Exception(f"{name} not a valid model name")
+
+
+def _cost_function(
+    poly: NDArray[5],
+    halcon_parameters: HalconParameters,
+    opencv_parameters: OpenCVParameters,
+    pixels: NDArray[Any, Any, 2],
+) -> float:
+    halcon_parameters.Poly = poly
+    halcon_pixels_undistorted = halcon_parameters.undistorted_pixels(pixels=pixels)
+    opencv_pixels_undistorted = opencv_parameters.undistorted_pixels(pixels=pixels)
+    return np.sqrt(np.mean(np.linalg.norm(halcon_pixels_undistorted - opencv_pixels_undistorted, axis=2) ** 2))
+
+
+def _create_pixels(zivid_camera_parameters: ZividCameraParameters, samples_x: float = 25) -> NDArray[Any, 1, 2]:
+    x = np.linspace(0, zivid_camera_parameters.image_size[0] - 1, samples_x)
+    y = np.linspace(
+        0,
+        zivid_camera_parameters.image_size[1] - 1,
+        round(samples_x * zivid_camera_parameters.image_size[1] / zivid_camera_parameters.image_size[0]),
+    )
+    XX, YY = np.meshgrid(x, y)
+    return np.array([[XX.flatten(), YY.flatten()]]).transpose((2, 0, 1))
+
+
+def _estimate_halcon_parameters_from_opencv(
```

Comment:
Since `samples_x` is an optional it feels like this is a function that is intended for the user. If that's the case, then make it public (remove preceding _), or make samples_x a constant in the function. If the function is made public then add a docstring.

Reply:
Done